src/parser/jobs/drg/modules/BattleLitany.tsx

```diff
@@ -0,0 +1,232 @@
+	static title = t('drg.battlelitany.title')`Battle Litany`
+	static displayOrder = DISPLAY_ORDER.BATTLE_LITANY
+
+	@dependency private timeline!: Timeline
+
+	private history: BLWindow[] = []
+	private lastLitFalloffTime: number = 0
+
+	protected init() {
+		this.addEventHook('normalisedapplybuff', {to: 'player', abilityId: STATUSES.BATTLE_LITANY.id}, this.tryOpenWindow)
+		this.addEventHook('normalisedapplybuff', {by: 'player', abilityId: STATUSES.BATTLE_LITANY.id}, this.countLitBuffs)
+		this.addEventHook('removebuff', {to: 'player', abilityId: WINDOW_STATUSES}, this.tryCloseWindow)
+		this.addEventHook('cast', {by: 'player'}, this.onCast)
+	}
+
+	private countLitBuffs(event: NormalisedApplyBuffEvent) {
+		// Get this from tryOpenWindow. If a window wasn't open, we'll open one.
+		const lastWindow: BLWindow | undefined = this.tryOpenWindow(event)
+
+		// Find out how many players we hit with the buff.
+		// BL has two normalized windows? seems weird...
+		if (lastWindow) {
+			lastWindow.playersBuffed += event.confirmedEvents.filter(hit => this.parser.fightFriendlies.findIndex(f => f.id === hit.targetID) >= 0).length
+		}
+	}
+
+	private tryOpenWindow(event: NormalisedApplyBuffEvent): BLWindow | undefined {
+		const lastWindow: BLWindow | undefined = _.last(this.history)
+
+
+		if (lastWindow && !lastWindow.end) {
+			return lastWindow
+		}
+
+		if (event.sourceID && event.sourceID === this.parser.player.id) {
+			const newWindow = new BLWindow(event.timestamp)
+
+			// Handle multiple drg's buffs overwriting each other, we'll have a remove then an apply with the same timestamp
+			// If that happens, mark the window and return
+			newWindow.containsOtherDRG = this.lastLitFalloffTime === event.timestamp
+
+			this.history.push(newWindow)
+			return newWindow
+		}
+
+		return undefined
+	}
+
+	private tryCloseWindow(event: BuffEvent) {
+		// for determining overwrite, cache the status falloff time
+		this.lastLitFalloffTime = event.timestamp
+
+		// only track the things one player added
+		if (event.sourceID && event.sourceID !== this.parser.player.id)
+			return
+
+		const lastWindow: BLWindow | undefined = _.last(this.history)
+
+		if (!lastWindow) {
+			return
+		}
+
+		// Cache whether we've seen a buff removal event for this status, just in case they happen at exactly the same timestamp
+		lastWindow.buffsRemoved.push(event.ability.guid)
+
+		if (this.isWindowOkToClose(lastWindow)) {
+			lastWindow.end = event.timestamp
+		}
+	}
+
+	// Make sure all applicable statuses have fallen off before the window closes
+	private isWindowOkToClose(window: BLWindow): boolean {
+		if (!window.buffsRemoved.includes(STATUSES.BATTLE_LITANY.id)) {
+			return false
+		}
+		return true
+	}
+
+	private onCast(event: CastEvent) {
+		const lastWindow: BLWindow | undefined = _.last(this.history)
+
+		// If we don't have a window, bail
+		if (!lastWindow) {
+			return
+		}
+
+		const action = getDataBy(ACTIONS, 'id', event.ability.guid) as TODO
```

Comment:
This should be using the data dependency for `getAction(event.ability.guid)` which returns an `Action` type.